main.py

```diff
@@ -1,32 +1,94 @@
+        memgraph_port: int,
+        exporter_port: int,
+        pull_frequency_seconds: int,
+    ) -> None:
+        self._memgraph_endpoint_url = memgraph_endpoint_url
+        self._memgraph_port = memgraph_port
+        self._exporter_port = exporter_port
+        self._pull_frequency_seconds = pull_frequency_seconds
+
+    @classmethod
+    def from_yaml_file(cls, file_name: str = "config.yaml") -> "Config":
+        with open(file_name) as f:
+            data = yaml.load(f, Loader=SafeLoader)
+            return Config(
+                data[ConfigConstants.MEMGRAPH][ConfigConstants.ENDPOINT_URL],
+                data[ConfigConstants.MEMGRAPH][ConfigConstants.PORT],
+                data[ConfigConstants.EXPORTER][ConfigConstants.PORT],
+                data[ConfigConstants.GENERAL][ConfigConstants.PULL_FREQUENCY_SECONDS],
+            )
+
+    @property
+    def exporter_port(self) -> int:
+        return self._exporter_port
+
+    @property
+    def memgraph_endpoint_url(self) -> str:
+        return self._memgraph_endpoint_url
+
+    @property
+    def memgraph_port(self) -> int:
+        return self._memgraph_port
+
+    @property
+    def pull_frequency_seconds(self) -> int:
+        return self._pull_frequency_seconds
+
+
+config_singleton = None
 
-# Create a metric to track time spent and requests made.
-REQUEST_TIME = Summary('request_processing_seconds', 'Time spent processing request')
-SIZE_CTR = Counter('size_counter', 'Size counter description')
-SIZE_GG = Gauge('size_gauge', 'Size gauge description')
-HISTOGRAM = Histogram('histie', 'Histogram description')
 
-@REQUEST_TIME.time()
 def pull_metrics():
-    """ Pull Memgraph metrics """
-    res = requests.get("http://localhost:9092/")
+    # We first want to fetch metrics from the Memgraph's HTTP endpoint
+    res = requests.get(
+        f"{config_singleton.memgraph_endpoint_url}:{config_singleton.memgraph_port}"
+    )
+
+    print(f"Pulled metrics: status code {res.status_code}")
     if res.status_code != 200:
-        raise Exception("Status code is not 200!")
-    
+        raise Exception(
+            f"Status code is not 200, but {res.status_code}, please check running services!"
+        )
+
     json_data = res.json()
-    size = json_data["size"]
-    SIZE_CTR.inc(size)
-    SIZE_GG.set(size)
-    HISTOGRAM.observe(random.random())
 
+    update_metrics(json_data)
+
+
+if __name__ == "__main__":
+    # Parse the configuration for starting the service and retrieve data from correct endpoints
+    config_singleton = Config.from_yaml_file()
+    start_http_server(config_singleton.exporter_port)
 
-if __name__ == '__main__':
-    # Start up the server to expose the metrics.
-    start_http_server(8000)
-    # Generate some requests.
+    # Continuously fetch metrics
     while True:
-        time.sleep(5)
-        pull_metrics()
+        time.sleep(config_singleton.pull_frequency_seconds)
+        try:
+            pull_metrics()
```

Comment:
this is wrong as if the first method raises an error it would ping more quickly and not sleep at all. I will set it before pull metrics but in the try block.